### PR TITLE
Move crm id store

### PR DESCRIPTION
**Pull request by nissamai**
The storage of CRM user identification information in our database is currently being handled by the class that interfaces with the CRM. As this coupling is unnecessary and also makes migrating our CRM difficult, this PR moves handling client-side (sefaria server) data storage to its own class (CrmInfoStorage), and adds another class (CrmMediator) to handle the coordination of information passing between the class that reads and writes to the CRM and the class that handles information storage about access to objects in the CRM so that the mediator class can be directly called by views that want to perform a CRM-related action.

This PR should keep NB code functionality intact. Salesforce methods are not yet implemented (outside of the create-new-user method).

It is a known problem that subscription lists (which will be handled differently once we migrate) are coupled with Sefaria front-end code. This will be handled in the next PR.

Feel free to comment on this PR or on the card (same information on the card as in PR), and LMK if you have any questions. Thanks! https://trello.com/c/lIX8KUBY

### CODE REVIEW

File: sefaria/helper/crm/crm_connection_manager.py [hunk 1]
```diff
@@ -42,3 +43,23 @@ def mark_as_spam_in_crm(self, crm_id):
         Do in CRM whatever decision has been made to do about spam users.
         """
         pass
+
+    def subscribe_to_lists(self, email, first_name=None, last_name=None, lang="en", educator=False):
+        CrmConnectionManager.validate_email(email)
+        CrmConnectionManager.validate_name(first_name)
+        CrmConnectionManager.validate_email(last_name)
```
Comment (relyks) [inline sefaria/helper/crm/crm_connection_manager.py:50]: Name isn't an email

**Review by relyks — CHANGES_REQUESTED** [review]
Some things need to be addressed but some of what I commented are just suggestions. If your tests work, I think everything is good and easier to understand than before.

File: sefaria/helper/crm/crm_info_store.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                user_profile.save()
+                return True
+            return False
+        elif crm_type == "SALESFORCE":
+            user_profile = UserProfile(email=email, user_registration=True)
+            if user_profile.id is not None and user_profile.sf_app_user_id != profile_id:
+                user_profile.sf_app_user_id = profile_id
+                user_profile.save()
+                return True
+            return False
```
[the remark this answers is not in the document]
Comment (nissamai) [reply sefaria/helper/crm/crm_info_store.py:26]: hopefully not, my sense is it's probably not worth it to do right now

File: sefaria/helper/crm/salesforce.py [hunk 3]
```diff
@@ -47,21 +53,37 @@ def _get_connection(self):
...
+        except:
+            # log
+            return False
+        return res
+
 
-    def test_get(self):
-        pass
+    def change_user_email(self, lists, email):
+        return self.put()
```
[the remark this answers is not in the document]
Comment (nissamai) [reply sefaria/helper/crm/salesforce.py]: nope, this is just not implemented but I did not mark it.

File: sefaria/helper/crm/crm_info_store.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    @staticmethod
+    def get_crm_id(uid=None, email=None, profile=None, crm_type=sls.CRM_TYPE):
+        if profile:
+            user_profile = profile
+        elif email:
+            user_profile = UserProfile(email=email)
+        elif uid:
+            user_profile = UserProfile(id=uid)
+        if crm_type == "NATIONBUILDER":
```
[the remark this answers is not in the document]
Comment (nissamai) [reply sefaria/helper/crm/crm_info_store.py:38]: confirmed via convo that you mean newline: will add that.

looking at this again I'm realizing that if no uid, email, or profile is provided this just silently returns nothing, so I'm going to raise an error in that case. thanks skyler!

**Pull request merged**